File: Cpp/204_Count_Primes.cpp
class Solution {
public:
    int countPrimes(int n) {
        int result = 0;
        vector<bool> prime(n+1, true);
        for(int i=2; i<n; ++i){
            if(!prime[i]) continue;
            for(int j=2; j*i<=n; ++j){
                prime[i*j] = false;
            }
        }
        
        for(int i=2; i<n; ++i){
            if(prime[i]) ++result;
        }
    
        return result;
    }
};

File: Kotlin/204_Count_Primes.kt
class Solution {
    fun countPrimes(n: Int): Int {
        val prime = BooleanArray(n + 1) { true }
        
        for(i in 2..n) {
            if(!prime[i]) continue
            
            for(j in 2 * i..n step i) {
                prime[j] = false
            }
            
        }
        
        return prime.filterIndexed { index, _ -> index > 1 && index < n }
                    .count { it }
    }
}